test/controllers/redirect.controller.test.ts

```diff
@@ -199,7 +198,53 @@ describe('RedirectController', () => {
       await request(app)
         .get(REDIRECT_GATE_URI)
         .expect(MOVED_TEMPORARILY)
-        .expect('Location', ROOT_URI + '/')
+        .expect('Location', CONFIRMATION_URI)
+    })
+  })
+
+  describe('payment fallback GET request', () => {
+    it('should redirect to confirmation page if GovPay state is valid and status is paid', async () => {
+      const dissolutionSession: DissolutionSession = generateDissolutionSession()
+
+      const app = createApp(container => {
+        container.rebind(SessionService).toConstantValue(instance(session))
+        container.rebind(DissolutionService).toConstantValue(instance(service))
+      })
+
+      dissolutionSession.paymentStateUUID = 'ABC123'
+
+      when(session.getDissolutionSession(anything())).thenReturn(dissolutionSession)
+
+      await request(app)
+        .get(PAYMENT_CALLBACK_URI)
+        .query({
+          state: 'ABC123',
+          status: 'paid'
+        })
+        .expect(MOVED_TEMPORARILY)
+        .expect('Location', CONFIRMATION_URI)
+    })
+
+    it('should redirect to not found if GovPay state is invalid or status is not paid', async () => {
```

Comment:
i think this can be 2 tests:
1. it should redirect to not found if gov pay state is invalid and status is paid
2. it should redirect to not found if gov pay state is valid and status is not paid